pkg/skaffold/runner/runcontext/context.go

```diff
@@ -110,6 +110,18 @@ func (ps Pipelines) TestCases() []*latest.TestCase {
 	return tests
 }
 
+func (ps Pipelines) StatusCheck() *bool {
+	var sc *bool
+	// set the group status check to disabled if any pipeline has StatusCheck
+	// set to false.
```

Comment:
I was not sure what would be reasonable here; how/why would a user set only a single pipeline's `StatusCheck` value to false? (Not 100% clear on the relationship between a skaffold.yaml file, a profile, and a pipeline, to be honest...)

Reply:
I'm not completely sure the reason a user would disable it in one, @gsquared94 might know a situation tho.

a skaffold.yaml can contain one or more skaffod configs. A config defines a pipeline so that skaffold knows how to build, test, deploy, and do statuscheck for a project. A config can also define profiles that replace parts of main config when the user activates a profile

Reply:
> a skaffold.yaml can contain one or more skaffod configs. A config defines a pipeline 

Ah, that makes sense. Thank you for clarifying! Let me know if it would be more reasonable to (1) disable healthcheck if any pipeline has StatusCheck set to false, or (2) enable healthcheck unless all pipelines have StatusCheck set to false. Thanks again for all your help with this!

Reply:
had a small discussion on this out of band, but this is probably a new scenario where we want to have values set on a per-config basis. there are surely situations where users want to enable health check for a single module, but leave it disabled for others (e.g. if one module is deploying something long running background task, health checks aren't useful), so it seems strange to disable it altogether if one particular module wants it off.

looking at how we have the code structured around the `RunContext` right now, this isn't a trivial change, so in interest of not completely blocking this implementation I think we can do this in a follow up PR. in the meantime, we should do two things here:

1) disable the health check if any pipelines explicitly disable it, since the alternative would be enabling it for long-running deployments that could cause frustrating UX
2) error out if users are explicitly setting status check configuration in one pipeline, and explicitly disabling it in any other. this will prevent surprising behavior, and at least give users a thread they can follow to get things to work in a less confusing way (we should probably point to docs)

@maggieneterval would you mind also creating an issue to track the follow up work here, and tie it back to this PR? something like "Support configuring status check independently between modules"

Reply:
sorry I was out for a few days, but @nkubala's point above is exactly what I wanted to say.

Reply:
Thank you both so much for your help with this. I opened https://github.com/GoogleContainerTools/skaffold/issues/5746 to document this follow-up work.